### DISCUSSION TITLE
Get http uri from environment variable

### DISCUSSION THREAD

**Discussion by janiu-001** [vectordotdev/vector#17126] [Q&A] [answered]
Hi expert,

Now we use http server as our sinks, is it possible to get the buffer size and uri from the environment variable? The uri should be different for our testing  and production!

<img width="829" alt="image" src="https://user-images.githubusercontent.com/19970581/231481485-b184fb9b-d9af-45cc-a36e-db74bea2423e.png">

**Comment by spencergilbert** [comment, marked as answer]
Hi @janiu-001 did you see [this section of the documentation](https://vector.dev/docs/reference/configuration/#environment-variables) on using environment variables in your configuration?

**Reply by janiu-001** [reply]
Yes, but it is only for "transform", right? It can not work in source and sinks, right?

<img width="792" alt="image" src="https://user-images.githubusercontent.com/19970581/231542452-120e98a6-3a20-4697-a9d4-dfd53565daa5.png">

**Reply by jszwedko** [reply]
Hey @janiu-001 ! No, you can use environment variables anywhere in the configuration file.

**Reply by janiu-001** [reply]
Thanks, is it possible to contcat the string? like 

<img width="687" alt="image" src="https://user-images.githubusercontent.com/19970581/231555228-adf684fe-0002-4f34-8c5e-28dbe56a700e.png">

**Reply by jszwedko** [reply]
Did you try and run into a problem?

**Reply by janiu-001** [reply]
it work well, thanks
